android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt

```diff
@@ -171,10 +171,12 @@ constructor(
 
     if (carePlanModified) saveCarePlan(output)
 
-    fhirResourceUtil.updateUpcomingTasksToDue(
-      subject = subject.asReference(),
-      taskResourcesToFilterBy = carePlanTasks,
-    )
+    if (carePlanTasks.isNotEmpty()) {
+      fhirResourceUtil.updateUpcomingTasksToDue(
+        subject = subject.asReference(),
+        taskResourcesToFilterBy = carePlanTasks,
+      )
+    }
```

Comment:
is this for efficiency reasons? If we did call it, would it not be a problem, but it would always be a no-op?

Wondering if we should push this check into the function

Reply:
Yah, I think we have no reason to call the method if the task list is empty.

Reply:
OK after reading `updateUpcomingTasksToDue` it does not make sense to push this check in and I get why this is a special case since it's being called only after tasks just generated. I also can't see an obvious way to test this now